src/scaffoldmaker/utils/tubenetworkmesh.py

```diff
@@ -250,19 +255,163 @@ def getElementsCountAcrossTransition(self):
+                endCurveLocation = (rawElementsCountAlong, 1.0)
+                endLength = lx[-1][0]
+                if endTrimSurface:
+                    surfacePosition, curveLocation, intersects = endTrimSurface.findNearestPositionOnCurve(cx, cd2)
+                    if intersects:
+                        endCurveLocation = curveLocation
+                        endLength = evaluateCoordinatesOnCurve(lx, ld, endCurveLocation)[0]
+                endCurveLocations.append(endCurveLocation)
+                endLengths.append(endLength)
+                sumEndLengths += endLength
+
+        minStartLength = min(startLengths)
+        maxStartLength = max(startLengths)
+        minEndLength = min(endLengths)
+        maxEndLength = max(endLengths)
+        maxLength = maxEndLength - minStartLength
+        # minimum number applies to fixedElementsCountAlong and targetElementLength
         minimumElementsCountAlong = 2 if (self._isLoop or ((self._junctions[0].getSegmentsCount() > 2) and
-                (self._junctions[1].getSegmentsCount() > 2))) else 1
-        elementsCountAlong = None
+                                                           (self._junctions[1].getSegmentsCount() > 2))) else 1
+        # small fudge factor on targetElementLength so whole numbers chosen on centroid don't go one higher:
+        elementsCountAlong = max(minimumElementsCountAlong, fixedElementsCountAlong if fixedElementsCountAlong else
+            math.ceil(maxLength * 0.9999 / targetElementLength))
+        startTransitionSize = transitionFactor * (maxStartLength - minStartLength)
+        endTransitionSize = transitionFactor * (maxEndLength - minEndLength)
+        if (startTransitionSize + endTransitionSize) > maxLength:
+            # reduce transitions in proportion to fit mean length
+            startTransitionSize *= maxLength / (startTransitionSize + endTransitionSize)
+            endTransitionSize = maxLength - startTransitionSize
+        maxElementLength = maxLength / elementsCountAlong
+        # print("maxLength", maxLength, "maxElementLength", maxElementLength)
+        # for parametric coordinate [0,1] over startTransitionSize, these are the start derivatives
+        # where mean would have value startTransitionSize at both ends
+        startTransitionXiSpacing = (maxElementLength / startTransitionSize) if (startTransitionSize > 0.0) else 0.0
+        endTransitionXiSpacing = (maxElementLength / endTransitionSize) if (endTransitionSize > 0.0) else 0.0
+        dStart = [[None] * self._elementsCountAround for _ in range(self._pathsCount)]
+        dEnd = [[None] * self._elementsCountAround for _ in range(self._pathsCount)]
+        startTransitionEndLength = minStartLength + startTransitionSize
+        endTransitionStartLength = maxEndLength - endTransitionSize
+        # print("Transition start min", minStartLength, "max", maxStartLength, "size", startTransitionSize, "end", startTransitionEndLength)
```

Comment:
Just a reminder to clean up commented print statements if no longer needed.

Reply:
Normally I'd remove them, but I'll wait until these changes are well-accepted first.